Topics/Tech_Stacks/FastAPI.md

```diff
@@ -0,0 +1,106 @@
+# FastAPI
+
+## Table of Contents
+### [Introduction](#introduction-1)
+### [Writing APIs](#writing-apis-1)
+### [Testing](#testing-1)
+
+
+## Introduction
+
+FastAPI is a popular web framework in Python for writing REST APIs. It is fast and easy to use, with type hints and automatic data validation. 
+
+It also auto generates documentation for your APIs with the [OpenAPI](https://www.openapis.org/) specifications and with an UI such as [Swagger UI](https://swagger.io/)
+
+You can check out FastAPI's tutorials [here](https://fastapi.tiangolo.com/tutorial/) at their own website with instructions on how to install FastAPI and how to get started with your first REST API using FastAPI
+
+Or if you wish to browse their documentation at your leisure, click [here](https://fastapi.tiangolo.com/)
+
+
+## Writing APIs
+
+The FastAPI tutorial on writing your first API from [here](https://fastapi.tiangolo.com/tutorial/first-steps/) is a very good starting point. It shows that all your API files will look something like this:
+
+``` {python}
+from fastapi import FastAPI
+
+app = FastAPI()
+
+# your APIs here
+@app.get("/test")
+def test():
+  return {"test message": "hello world"}
+
+```
+
+However, it is highly likely that not all your APIs are going to be in the same file. For example, take the above as your main.py. You are likely not going to put all your APIs there, so you may create a file called
+more_apis.py.
+
+To connect more_apis.py with your main.py, you will need to use routers like this:
+
+main.py
+``` {python}
+from fastapi import FastAPI
+import more_apis  # change this import to whatever file you create and to whatever location you created it at
+
+app = FastAPI()
+app.include_router(more_apis.router) # your router connecting more_apis.py APIs
+
+@app.get("/test")
+def test():
+  return {"test message": "hello world"}
+
+```
+
+more_apis.py
+``` {python}
+from fastapi import APIRouter
+
+router = APIRouter()
+
+# your other APIs
+@router.get("/otherAPI")
+def other_API():
+  return {"different message": "hello world again"}
+
+```
+
+You can see more details about FastAPi and routers [here](https://fastapi.tiangolo.com/tutorial/bigger-applications/)
+
+As always, you can check the official FastAPI tutorial [here](https://fastapi.tiangolo.com/tutorial/) for anything about writing APIs.
+
+
+## Testing
+
+Testing with FastAPI can be automated with GitHub actions and uses FastAPI's TestClient and [pytest](https://docs.pytest.org/en/7.4.x/) to test. 
```

Comment:
I think that when mentioning the TestClient, it could be helpful to briefly explain why it's used for testing, or link to documentation about it, such as https://fastapi.tiangolo.com/reference/testclient/). I think this would allow the reader to get a better idea of classes and terms that may be completely unfamiliar to them, if they so choose.

Also, just a nitpick, but I think it should be "Github Actions" instead of "Github actions".

Reply:
Added and talked about test client